### PR TITLE
CSP realtime errors

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This does a couple of things
(1) Turns on undef in the linter, which means we now get warnings if we use a variable without declaring it
(2) Take our list of functions and mark them as predef'ed so that we don't get warnings about them
(3) Map some of the error strings that we get to new strings
(4) Extract this mapping logic into its own module. This allows for easier testing. I also like the idea of minimizing how much code I write in mode-javascript_codeorg.js, where I'm trying to have code follow the ace code style.
(5) Moves mode file and error mapping into subdirectory within applab.

### CODE REVIEW

File: apps/src/applab/acemode/errorMapper.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * Takes the results of a JSLint pass, and modifies the error text according to
+ * our mapping. Note this makes changes in place to the passed in results
+ * object.
+ */
+module.exports.processResults = function (results) {
+  results.data.forEach(function (item) {
+
+    errorMap.forEach(function (errorMapping) {
+      var old = errorMapping[0];
+      var replacement = errorMapping[1];
```
Comment (bcjordan) [inline apps/src/applab/acemode/errorMapper.js]: These 2 vars unused?

Comment (Bjvanminnen) [inline apps/src/applab/acemode/errorMapper.js]: Yep. From an earlier iteration. Ironic, considering the change I'm making. I bet jshint complains if I run it.

**Comment by bcjordan** [comment]
LGTM. Nice testing & good call moving into a separate module

File: apps/src/applab/acemode/errorMapper.js (hunk 1 above)
Comment (Bjvanminnen) [inline apps/src/applab/acemode/errorMapper.js]: Surprisingly, it does not.

**Pull request merged**